### PR TITLE
Migrate snippet form to formik

### PR DESCRIPTION
**Pull request by kulyk**
Epic #26178

1. Migrates `SnippetModal` to `formik`
2. Breaks `SnippetModal` into `SnippetForm` and `SnippetModalForm`
3. Covers both modal and form with unit-tests
4. Removes form-related code from the snippets entity

## To Verify

> **Note**
> Form behavior can be different depending on if you're running an OSS or paid version of Metabase. Paid Metabase plans allow to create snippet folders (called snippet collections internally) to organize snippets. So for OSS, you shouldn't see a folder picker at all. In paid plans, you should see one **if you've created at least one folder**

### OSS

1. New > SQL question
2. Click the "snippets" icon at the right side of the query content editor (right above the "Run" button)
3. Click "Create snippet"
4. Ensure you can't submit the form until both name and content are filled in
5. Ensure you can save and use the snippet
6. Once you've created at least one snippet, click on it in the snippets sidebar and then click "Edit"
7. Ensure you can rename and change content/description for a snippet
8. Ensure you can archive the snippet using the "Archive" button at the bottom left of the snippet editing modal (also make sure you can't see this button when creating a new snippet)

### Paid

1. Spin up paid Metabase and sign in as admin
2. New > SQL question
3. Click the "snippets" icon at the right side of the query content editor (right above the "Run" button)
4. Click the "+" icon at the top right of the snippets sidebar and pick "new folder"
5. Create a folder
6. Test out both snippet creating and editing flows and ensure you can pick a snippet folder now

## Demo

https://user-images.githubusercontent.com/17258145/205351179-992ec560-e9dd-473a-9acf-af4ff95f6eab.mp4

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.styled.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import styled from "@emotion/styled";
+import FormTextArea from "metabase/core/components/FormTextArea";
+import { color } from "metabase/lib/colors";
+
+export const FormSnippetTextArea = styled(FormTextArea)`
+  textarea {
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.styled.tsx:6]: `${TextArea.Root} {`? We might need to add `Root` similar to other components

File: frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+          />
+          <FormInput
+            name="name"
+            title={t`Give your snippet a name`}
+            placeholder={t`Current Customers`}
+          />
+          <FormInput
+            name="description"
+            title={t`Add a description`}
+            placeholder={t`It's optional but oh, so helpful`}
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx:172]: `nullable`?

**Review by ranquild — APPROVED** [review]

File: frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  onArchive?: () => void;
+  onCancel?: () => void;
+}
+
+interface SnippetLoaderProps {
+  snippetCollections: Collection[];
+}
+
+interface SnippetFormDispatchProps {
+  handleCreateSnippet: (
```
[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx:69]: @ranquild, we already have an `onCreate` prop that runs _after_ the snippet is created and this one is actually creating the snippet. Maybe `handleCreate` isn't the best choice of a name here?

File: frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx (hunk 2 above)
Comment (kulyk) [reply frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx:172]: Right, I always forget about this, thanks

File: frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx:69]: Maybe `onCreate` and `onCreateSuccess`? I've seen this in other our components

[the remark this answers is not in the document]
Comment (kulyk) [reply frontend/src/metabase/query_builder/components/template_tags/SnippetForm/SnippetForm.tsx:69]: Sounds good. Mind if I merge that as is an open another PR that'd also fix that in other forms?

**Pull request merged**